tests/test-map-names

```diff
@@ -0,0 +1,73 @@
+#!/bin/bash
+# SPDX-License-Identifier: BSD-3-Clause
+#
+# test-map-names - Test that maps have the expected names.
+#
+# This script tests that the map names (the name such as MAP05, E2M3, etc., as
+# indicated by the name of the first lump for the map) match the name of the
+# file they're in. It is expected that WAD files in the "levels" directory will
+# match in this way after certain translations (CxMy to ExMy, etc.).
+
+# Globals
+
+this_dir="${0%/*}"                 # Directory that contains this script.
+top_dir=$(realpath "$this_dir/..") # Top level directory.
+levels_dir="$top_dir/levels"       # The "levels" directory with WADs to check.
+boot_dir="$top_dir/bootstrap"      # The "bootstrap" directory.
+format="%-12s%-12s%-12s\n"
+
+if ! cd "$levels_dir"
+then
+    echo "Could not CD to levels directory \"$levels_dir\"." 1>&2
+    exit 1
+fi
+
+errors=0
+unset header_shown
+for wad in c*.wad dm*.wad map*.wad
+do
+    # What we expect the map name to be based on the filename.
+    expected_name=${wad%.wad}
+    expected_name=${expected_name^^}
+    expected_name=${expected_name/DM/MAP}
+    expected_name=${expected_name/C/E}
+    if [[ -z $expected_name ]]
+    then
+        echo "Could not determine the expected name for WAD \"$wad\"." 1>&2
+        exit 1
+    fi
+
+    # What the map name actually is as indicated by the name of the first lump
+    # for the map.
+    actual_name=$(deutex -doom2 "$boot_dir" -wadir "$wad" | \
```

Comment:
Probably this is sufficient:

if ! deutex -wadir "$wad" | grep -qi "$expected_name"; then
  ...

Reply:
I just added a top level comment showing what the output looks like, which includes the `actual_name`. Admittedly the `actual_name` is obvious to people familiar with this project, but it's a way for people to be told exactly what the map name should be. It's also a way to make sure this script is doing the right thing.  So that's why I'm grabbing it explicitly.

Also, I couldn't get it to work without the `-doom2` option.